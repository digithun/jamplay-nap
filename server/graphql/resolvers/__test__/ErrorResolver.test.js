/* eslint-env jest */
describe('ErrorResolver', () => {
  it('should return null if no error', () => {
    const ErrorResolver = require('../ErrorResolver')
    const context = { nap: { errors: [] } }
    expect(ErrorResolver.resolver({ context })).toMatchSnapshot()
  })

  it('should return errors if has any error', () => {
    const ErrorResolver = require('../ErrorResolver')
    const { onError } = require('../../../errors')
    const context = { nap: { errors: [] } }

    try {
      onError(context)(new Error('foo'))
    } catch (err) {
      expect(err).toMatchSnapshot()
    }

    expect(ErrorResolver.resolver({ context })).toMatchSnapshot()
  })
})
