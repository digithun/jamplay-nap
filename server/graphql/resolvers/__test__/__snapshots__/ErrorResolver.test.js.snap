// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorResolver should return errors if has any error 1`] = `[Error: foo]`;

exports[`ErrorResolver should return errors if has any error 2`] = `[Function]`;

exports[`ErrorResolver should return null if no error 1`] = `[Function]`;
